src/network/server.rs

```diff
@@ -420,14 +420,48 @@ impl HubService for MyHubService {
 
         let mut message = request.into_inner();
         message_bytes_decode(&mut message);
-        let response_message = self.submit_message_internal(message, false).await?;
+        let fid = message.fid();
+        let msg_type = message.msg_type().into_i32();
+        let result = self.submit_message_internal(message, false).await;
 
         self.statsd_client.time(
             "rpc.submit_message.duration",
             start_time.elapsed().as_millis() as u64,
         );
 
-        Ok(Response::new(response_message))
+        match result {
+            Ok(message) => {
+                self.statsd_client.count("rpc.submit_message.success", 1);
+                Ok(Response::new(message))
+            }
+            Err(err) => {
+                self.statsd_client.count("rpc.submit_message.failure", 1);
+                info!(
+                    hash = hash,
+                    fid = fid,
+                    errCode = err.code,
+                    msgType = msg_type,
+                    "submit_message failed: {}",
+                    err
+                );
+                let err_code = err.code.as_str();
+                let mut status = if err_code.starts_with("bad_request") {
+                    Status::invalid_argument(err.to_string())
+                } else if err_code == "not_found" {
+                    Status::not_found(err.to_string())
+                } else if err_code.starts_with("db") || err_code.starts_with("internal") {
+                    Status::internal(err.to_string())
+                } else if err_code.starts_with("unavailable") {
+                    Status::unavailable(err.to_string())
+                } else {
+                    Status::unknown(err.to_string())
+                };
+                if let Ok(err_str) = AsciiMetadataValue::from_str(&err_code) {
+                    status.metadata_mut().insert("x-err-code", err_str);
```

Comment:
Hubble returns this directly as `errCode` but that's not valid with the rust library. I'll update hub-nodejs to read from this as well so clients get the error code populated correctly.

Reply:
https://github.com/farcasterxyz/hub-monorepo/pull/2567